prefect_dbt/cloud/jobs.py

```diff
@@ -630,3 +639,168 @@ def retry_dbt_cloud_job_run_subset_and_wait_for_completion_flow():
         poll_frequency_seconds=poll_frequency_seconds,
     )
     return run_data
+
+
+class DbtCloudJob(Block, ABC):
+    """
+    placeholder
+    """
+
+    _block_schema_capabilities = [
+        "trigger-job",
+        "wait-for-job-completion",
+        "fetch-job-results",
+    ]
+
+    dbt_cloud_credentials: DbtCloudCredentials
+    job_id: int
+    max_wait_seconds: int
+    poll_frequency_seconds: int
+
+    @property
+    def logger(self):
+        """
+        placeholder
+        """
+        try:
+            return get_run_logger()
+        except MissingContextError:
+            return get_logger()
+
+    @sync_compatible
+    async def trigger(self, **trigger_job_run_options):
+        """
+        placeholder
+        """
+        try:
+            async with self.dbt_cloud_credentials.get_administrative_client() as client:
+                response = await client.trigger_job_run(
+                    job_id=self.job_id, options=trigger_job_run_options
+                )
+                self.logger.info("Triggered job {self.job_id!r}")
+        except HTTPStatusError as ex:
+            raise DbtCloudJobRunTriggerFailed(extract_user_message(ex)) from ex
+
+        run_data = response.json()["data"]
+        run_id = run_data.get("id")
+        self.logger.info("Job run has ID {run_id!r}")
+        return run_id
+
+    @sync_compatible
+    async def get_run(self, run_id):
+        """
+        placeholder
+        """
+        try:
+            async with self.dbt_cloud_credentials.get_administrative_client() as client:
+                response = await client.get_run(run_id=run_id)
+        except HTTPStatusError as ex:
+            raise DbtCloudGetRunFailed(extract_user_message(ex)) from ex
+        run_data = response.json()["data"]
+        return run_data
+
+    @sync_compatible
+    async def get_job(self, order_by: Optional[str] = None):
+        """
+        placeholder
+        """
+        try:
+            async with self.dbt_cloud_credentials.get_administrative_client() as client:
+                response = await client.get_job(
+                    job_id=self.job_id,
+                    order_by=order_by,
+                )
+        except HTTPStatusError as ex:
+            raise DbtCloudGetJobFailed(extract_user_message(ex)) from ex
+        return response.json()["data"]
+
+    @sync_compatible
+    async def get_status(self, run_id) -> int:
+        """
+        placeholder
+        """
+        run_data = self.get_run_data(run_id)
+        run_status_code = run_data.get("status")
+        return run_status_code
+
+    def terminal_state_reached(self, run_status_code) -> bool:
+        return DbtCloudJobRunStatus.is_terminal_status_code(run_status_code)
```

Comment:
if this is a one-liner, do you think this needs to be a method of the class? 

are you thinking `terminal_state_reached` should be implemented by all types of `JobBlock`?

Reply:
nevermind, I'm encountering a similar need for ad-hoc methods on the job block